binding/ios/RhinoManager.swift

```diff
@@ -98,92 +77,54 @@ public class RhinoManager {
                 usleep(10000)
             }
-            self.audioInputEngine.stop()
+            VoiceProcessor.shared.stop()
             
             self.started = false
             self.stop = false
         }
         
-        try audioInputEngine.start()
+        try VoiceProcessor.shared.start(
+            frameLength: Rhino.frameLength,
+            sampleRate: Rhino.sampleRate,
+            audioCallback: self.audioCallback
+        )
      
         self.started = true        
-    }   
-}
-
-private class AudioInputEngine {
-    private let numBuffers = 3
-    private var audioQueue: AudioQueueRef?
-    
-    var audioInput: ((UnsafePointer<Int16>) -> Void)?
-    
-    func start() throws {
-        var format = AudioStreamBasicDescription(
-            mSampleRate: Float64(Rhino.sampleRate),
-            mFormatID: kAudioFormatLinearPCM,
-            mFormatFlags: kLinearPCMFormatFlagIsSignedInteger | kLinearPCMFormatFlagIsPacked,
-            mBytesPerPacket: 2,
-            mFramesPerPacket: 1,
-            mBytesPerFrame: 2,
-            mChannelsPerFrame: 1,
-            mBitsPerChannel: 16,
-            mReserved: 0)
-        let userData = UnsafeMutableRawPointer(Unmanaged.passUnretained(self).toOpaque())
-        AudioQueueNewInput(&format, createAudioQueueCallback(), userData, nil, nil, 0, &audioQueue)
-        
-        guard let queue = audioQueue else {
-            return
-        }
-        
-        let bufferSize = UInt32(Rhino.frameLength) * 2
-        for _ in 0..<numBuffers {
-            var bufferRef: AudioQueueBufferRef? = nil
-            AudioQueueAllocateBuffer(queue, bufferSize, &bufferRef)
-            if let buffer = bufferRef {
-                AudioQueueEnqueueBuffer(queue, buffer, 0, nil)
-            }
-        }
-        
-        AudioQueueStart(queue, nil)
     }
     
-    func stop() {
-        guard let audioQueue = audioQueue else {
+    /// Callback to run after after voice processor processes frames.
+    private func audioCallback(pcm: [Int16]) {
+        guard self.rhino != nil else {
             return
         }
-        AudioQueueFlush(audioQueue)
-        AudioQueueStop(audioQueue, true)
-        AudioQueueDispose(audioQueue, true)
-        audioInput = nil
-    }
-    
-    private func createAudioQueueCallback() -> AudioQueueInputCallback {
-        return { userData, queue, bufferRef, startTimeRef, numPackets, packetDescriptions in
-            
-            // `self` is passed in as userData in the audio queue callback.
-            guard let userData = userData else {
-                return
+
+        do {
+            let isFinalized:Bool = try self.rhino!.process(pcm: pcm)
+            if isFinalized {
+                do {
+                    let inference:Inference = try self.rhino!.getInference()
+                    self.onInferenceCallback?(inference)
+                } catch {
+                    print("There was an error retrieving the inference result.")
```

Comment:
I think we can probably eliminate the 'do { } catch { }' around getInference and handle it with the errorCallback below